### DISCUSSION TITLE
Search maps by author in game

### DISCUSSION THREAD

**Discussion by SoundBoomer** [ppy/osu#27101] [Ideas] [answered]
On the osu site, it's possible to search maps by their mapper by using author=""mappername"". Is it possible to transfer this into lazer as well? Sorting by author and scrolling all the way searching for the right letter is not that practical.

**Comment by brandondong** [comment]
I think there is a filter but it's `creator=...`
https://github.com/ppy/osu/blob/45b4edcd2325d51812849af63e150951cb675574/osu.Game/Screens/Select/FilterQueryParser.cs#L70

Weird that on the website, only `author` works and on lazer, only `creator` works... I feel like it should be changed so either or works on both.

**Comment by Stoppedpuma** [comment]
#27112

**Comment by bdach** [comment]
> On the osu site, it's possible to search maps by their mapper by using author=""mappername""

I don't think it is? I think you're confused and the fact that it works at all is a freak coincidence at best? `creator=` is the officially supported one.

@ppy/team-web am I wrong? Judging by [this](https://github.com/ppy/osu-web/blob/e9b555b7955649aac19e2db01eea083607c24280/app/Libraries/Search/BeatmapsetSearchRequestParams.php#L214-L233) it should be `creator=` right? People above might have been confused by guest difficulties or something because it will include these (so it'll show seemingly "unrelated" results with a different host mapper, but one of the difficulties _should_ be mapped by the searched user). `author=` working at all (because it does seem to work) is probably just elasticsearch tokenizer doing stuff to split the `author` thing off to a separate keyword or something?

**Reply by Stoppedpuma** [reply]
90% sure this is just confusion of author meaning generally the same thing, I've looked basically everywhere and could only find beatmap creators being referred to as authors in the wiki. I don't know how author came into mind here and it might be a good idea for me to either just update the wiki or submit another pr to osu!web since author is a general term, I'm sure I've tried searching by author at least a few times and this might be a good thing for new players or players that would refer to creator as author instead?

**Reply by bdach** [reply]
As stated in the PR thread, I'm not opposed to adding `author=`, I'm just generally confused by it being called a _client_ shortcoming specifically rather than just a feature request to all places that support these filters.

I just wanna make sure this doesn't mean that the web implementation of the filter needs tweaking wrt guest mappers or something. @SoundBoomer @brandondong please advise.

**Reply by brandondong** [reply]
I see, sorry about the confusion! I was mistaken about website behavior due to guest difficulty behavior. And yes, `author=` was only working for me due to keyword search, `author=vinxis` is the easiest way to disprove that.

**Reply by cl8n** [reply]
seems resolved but yeah `author` has no special meaning. if you want to add it, i can see "author" or "mapper" both being good synonyms, same change should be made to web as well

**Reply by SoundBoomer** [reply]
Thanks for clarifying this! In the game, there's a sort by "author" so i didn't know that there's another sort by "creator".

**Comment by Stoppedpuma** [comment, marked as answer]
Resolved by #27112
